compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -2985,12 +2985,10 @@ void OMR::X86::CodeGenerator::apply32BitLoadLabelRelativeRelocation(TR::Instruct
    {
    TR::Instruction *movInstructionX86   = movInstruction;
 
-
    TR_ASSERT(movInstructionX86->getOpCodeValue() == MOV4RegImm4,"wrong load instruction used for apply32BitLoadLabelRelativeRelocation\n");
 
    uint8_t *cursor = movInstructionX86->getBinaryEncoding();
-   cursor += movInstructionX86->rexPrefixLength();
-   cursor += movInstructionX86->getOpCode().getOpCodeLength();
+   cursor += movInstructionX86->getOpCode().length(movInstructionX86->rexBits()) + movInstructionX86->rexRepeatCount();
```

Comment:
rexBits() is always a single byte which contains only a necessary REX.
The addition of unnecessary 0x40 is counted as part of rexRepeatCount(), which returns the number of unnecessary REX bytes.